src/gluonnlp/model/bert.py

```diff
@@ -1310,17 +1333,25 @@ def get_roberta_model(model_name=None, dataset_name=None, vocab=None, pretrained
         Whether to include attention weights of each encoding cell to the output.
     output_all_encodings : bool, default False
         Whether to output encodings of all encoder cells.
+    ignore_extra : bool, default False
+        Whether to silently ignore parameters from the file that are not
+        present in this Block.
+    hparam_allow_override : bool, default False
+        If set to True, pre-defined hyper-parameters of the model
+        (e.g. the number of layers, hidden units) can be overriden.
 
     Returns
     -------
     RoBERTaModel, gluonnlp.vocab.Vocab
     """
-    predefined_args = bert_hparams[model_name]
-    mutable_args = ['use_residual', 'dropout', 'word_embed']
-    mutable_args = frozenset(mutable_args)
-    assert all((k not in kwargs or k in mutable_args) for k in predefined_args), \
-        'Cannot override predefined model settings.'
+    predefined_args = bert_hparams[model_name].copy()
+    if not hparam_allow_override:
+        mutable_args = ['use_residual', 'dropout', 'word_embed']
```

Comment:
Not changed in this PR, but why is `embed_size` not part of this?

Reply:
I copied some from the transformer get model function. Actually, i'm not sure whether we want to have a whitelist of mutable args. It make the fucntion harder to maintain, since we have `hparam_allow_override` flag.

Reply:
Do you want to remove the whitelist then?

Reply:
If so, let's do it in a separate PR to unblock the current one

Reply:
I'd prefer a separate one